### DISCUSSION TITLE
Quarg Expansion/Development

### DISCUSSION THREAD

**Discussion by OcelotWalrus** [OcelotWalrus/Cromha-Expansion-plugin#125] [General]
There is already a discussion on the Endless . Sky github but I want to expand the quarg in this plugin as well. We will talk about this here.

_Text who are marked ~like this~ means it has been done and fully finished_

_Check [Quarg Revamp Discussion](https://github.com/endless-sky/endless-sky/discussions/9016)_

_Check the Github Project [here](https://github.com/orgs/Cromha-Plugins/projects/4/views/1)_

We are now continuing the Quarg Revamp project, and I held a stream where we discussed most things overall and what choices we've made so far. [You can find a recording of that stream here.](https://www.youtube.com/watch?v=HJQO9wDVrLo) Below I will summarize what we've come to for a conclusion:
## Design Choices

Will update this with images once we get the design in a more completed state.

  * Skylark-esque - mostly dark greys, silvers, and perhaps small cases of glowing lights and light hues of blue-purple.

  * Bridge gap between Becca's and Todd's newest designs- not too spiked, not too curved. See the VOD linked above.

  * Avoid rocket ship shapes and uniform shape distribution.

  * Each Quarg ship class can have small differences in appearance based on their ringworld location - not necessarily stat-focused, but slightly altered sprites, like Archons.

 
## Outfit Choices

In addition to what is currently in the game, we will likely be pulling from the original [Quarg Expansion PR](https://github.com/endless-sky/endless-sky/pull/3665):

  * ~Fluxion Shield Generator~

  * ~Infimum Shield Generator~

  * ~Singularity Core~

  * ~Tachyon Core~

  * ~Central Repair Unit~

  * ~Peripheral Repair Unit~

  * ~Different-sized Graviton thrusters~

  * ~Blight Rifle (rifle for the Quarg)~

## Weapon Choices

No ballistics.

In addition to what is currently in the game:

  * ~Doomscythe~ - A weapon similar to the Prototype Fusion Cannon but way more powerful. It will be a turret.

  * Possible others being discussed.

## Ship Choices

No Fighters. No Transports. No Superheavies. Scale generally around Human ship level, depending on design and class. Names based on dragons/dragon mythology.

We've come up with a total of 10-11 ships for the Quarg's roster:
### Civilian Design

  * Lindwyrm Freighter (Skylark Replacement)

  * Amphithere Freighter

  * Kirin Freighter? (A Freighter designed for freight in times of conflict)

### Utility Design

  * Guivre Utility Ship (Ringworld construction ship - transports large loads of materials for them and assists in construction for them)

  * Smaug Freighter (highly defensible medium-sized freighter)

### Legacy Design (Older ships pre-Pug and Coalition Conflict)

  * Wardragon Light Warship

  * Drake Light Warship

  * Direwyrm Medium Warship

  * Glaurung Heavy Warship (their largest warship to be revealed in time of need, similar to Arfectia in use)
  * Ancalagon Heavy Warship (powerful capital ship, found normally in fleets)

### Recent Designs (Due to Pug and Coalition Conflict)

  * Wyvern Medium Warship

  * Hydra Medium Warship

  * Unnamed Heavy Warship (Appears as a rare spawn for the Korath Efret Quarg and during the Coalition Campaign)

  * Archdragon Heavy Warship (Only appears within the Coalition Campaign)

## Balance Notes

  * A Wardragon (LW) should not be able to take on a Heliarch Punisher. Two Wardragons have a better chance to take on a Punisher, but not easily.

  * A Quarg Medium Warship should beat 2 punishers relatively consistently; 3 Punishers should be likely to beat one.

  * A Quarg Heavy Warship should require at least 4-5 Punishers to defeat.

  * The Skylance needs to be nerfed. The original proposal is in half.

  * The Quarg should make use of active cooling more in their outfits (see the old PR).

  * The Antimatter Core needs to be rebalanced. (In turn, balance the other cores.)

## Suggestions

I'm leaving these unchecked until we have a general consensus on them being things we do/or we just decide on them anyway.

  * [x]  Perhaps Quarg thrusters can be given reverse thrust equal to their normal thrust.

  * [x]  Perhaps the Quarg should have a lower disable threshold. (they now all have their disable threshold at 25% of the hull)
  
  * [x] Perhaps the Quarg should have integrated or external after-burning systems 

  * [x]  Perhaps the Quarg should have lower armor, but more resistances.

Some ship and outfit names are obviously to be decided and are not final, others may be adjusted.

While design and overall ships and outfits/weapons have been decided upon, please use this opportunity to discuss balance as that's the category we need the most feedback in. Again, reference the older PR found [here](https://github.com/endless-sky/endless-sky/pull/3665) as a start. Thank you all for being so patient while we worked up to this point, and I can't wait to see what designs we look at for finals, as well as where the Quarg turn up in the relative future!

_Side-Note: Capn' Pester, the Author ship that uses the current Wardragon design, will keep that design. The name "Skylark" will no longer be used, however it will cause conflicts with pre-existing Quarg ships in saves if it is used elsewhere._

**Comment by lumbar527** [comment]
Suggestion: more ships not planned in the original discussion.

### Utility
- Smaug Freighter (highly defensible medium-sized freighter)

### Legacy
- Glaurung Heavy Warship (their largest warship to be revealed in time of need, similar to Arfectia in use)
- Ancalagon Heavy Warship (powerful capital ship, found normally in fleets)

**Reply by OcelotWalrus** [reply]
Looks good, I'm adding them!

**Comment by OcelotWalrus** [comment]
I've created the Quarg Voidscythe: 
![quarg voidscythe](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/afb38d34-9f04-49a4-aac7-84e888f6136b)
For now, it's just install on the `Quarg Hydra` but you should test the weapon in-game, it's a deadly one.

**Reply by lumbar527** [reply]
You could also post it on the Endless Sky discussion, though someone else might be working on a Voidscythe.

**Reply by OcelotWalrus** [reply]
Good idea!

**Reply by OcelotWalrus** [reply]
I actually, after thinking, this texture won't be good for the original discussion, it the original discussion, the Voidscythe is a beam turret, but this one is a projectile turret.

**Reply by lumbar527** [reply]
What if you changed it to be the Doomscythe or something, so it wouldn't conflict with Endless Sky one it's added?

**Reply by OcelotWalrus** [reply]
Great idea! Thanks for you help.

**Comment by OcelotWalrus** [comment]
We have a problem for the ships:

For half-part of the ships, we will use similar models like the Quarg Wardragon but for other we will use a model similar to these:
![quarg?](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/03da1464-b75f-44c8-bd9d-de695e4974cd)

![quarg??](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/f152542b-6584-4daa-8e2a-1d7f6bb50ab8)
![quarg???](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/1f8a88b5-d6cd-4e61-8974-96f63c18e003)
![quarg????](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/9a4696f7-ae7a-486a-8aea-5beab1ee9eb1)

But they don't give any blender/3D model of the ships. How are we supposed to create new ones?

**Reply by lumbar527** [reply]
Maybe you could ask on the original discussion, though that might take a while. Or you could just try to remake one, but that would be a lot more complicated.

**Reply by OcelotWalrus** [reply]
I think I'll ask and wait for a answer. While that I will create design like the Wardragon.

By the way, did you noticed the Quarg Wardragons textures are very similar to the Pug ships?

**Reply by lumbar527** [reply]
> By the way, did you noticed the Quarg Wardragons textures are very similar to the Pug ships?

I don't think I thought about that, but that's interesting.

**Reply by OcelotWalrus** [reply]
I think it's because the ship like the Wardragon, the Drak and the Hydra are past-Pug-war. And the others ships that looks more modern than organic are before-Pug-war. It might be the reason.

**Comment by OcelotWalrus** [comment]
Found the assets of the original Quarg Revamp PR:

https://drive.google.com/drive/folders/1qCnJMyfmY2Za6EFqBz2k4TqLFwHofdNS

**Reply by OcelotWalrus** [reply]
I'm going to download them and add it to the source/ directory.

**Reply by OcelotWalrus** [reply]
Done!

**Comment by OcelotWalrus** [comment]
Do you have any idea of what these could be?
![quantumium](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/5a2a09ba-5333-42d3-bf92-9aed50bd03d8)
![quarg cannonade](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/a3cee9be-c272-46c5-9224-57249f223b16)

**Reply by OcelotWalrus** [reply]
Maybe the second could be a special drive? A cloaking device?

**Reply by lumbar527** [reply]
First, what was it in the original plugin? (the second thing) Or, what about a Gravitational Core, which gives ships more thrust and turn, maybe some slowing resistance.
The first thing could be a rare ore-thing that the Quarg are the only ones who figured out how to use it.

**Reply by OcelotWalrus** [reply]
The second thing was originally attack/defense outfit, like the Countermeasure Module.

**Reply by OcelotWalrus** [reply]
I think the second texture will be a gravitational core thing like you said. It will be called the `Quarg Gravitational Device`.

**Comment by OcelotWalrus** [comment]
The `Quarg Gravitational Core`:
```
outfit "Quarg Gravitational Core"
	category "Systems"
	thumbnail "outfit/quarg gravitational device"
	"unplunderable"
	"slowing protection" 1
	"slowing resistance heat" .9
	"thrust" 50
	"turn" 600
	"force protection" 1
	"inertia reduction" 
	"inertia reduction" 10
	"drag reduction" .8
	"optical jamming" 35
	"radar jamming" 35
	"energy consumption" 60
	description "The Quarg developed the Gravitational Core is a device that generate gravitational forces to create more thrust and turn, enhance your ship acceleration in overhaul, it protects your ship from hit forces and slowing damages."
	description "It can also corrupt radar and optical tracking missiles by creating gravitational forces."
```

**Reply by OcelotWalrus** [reply]
It's really powerful.

**Reply by lumbar527** [reply]
Cool!

**Reply by OcelotWalrus** [reply]
I'm currently creating the Quarg Fafnir.

**Reply by OcelotWalrus** [reply]
I love the Quarg ships designs and Quarg in general.

**Reply by OcelotWalrus** [reply]
It's my favorite species.

**Reply by lumbar527** [reply]
Cool! I'm not sure I have a favorite Endless Sky species.

**Comment by OcelotWalrus** [comment]
Hey! Some news about the Quarg Expansion!
I've re-coded all outfits so they are no more problems about them and I added a few new Quarg ships: The Quarg Zilant, The Quarg Fafnir and the Quarg Hydra. You can checkout the [v0.0.3 Stable Release](https://github.com/Cromha-Plugins/Cromha-Expansion-plugin/releases/tag/v0.0.3.5)

**Reply by OcelotWalrus** [reply]
I've also made a specific Github Project to keep track of the development of the Quarg Expansion part of the plugin.
https://github.com/orgs/Cromha-Plugins/projects/4/views/1

**Comment by OcelotWalrus** [comment]
Just added the Quarg Kalisto and Quarg Drake ships!
Check commits d7f5df415e1a9f672826a3ac7ae8cd8cb125d671 and dafa6b2c7706e637a4ecf774a04a985a3e2f5939.
